backend/src/main/java/com/redhat/cloud/notifications/db/BehaviorGroupResources.java

```diff
@@ -41,14 +42,27 @@ public BehaviorGroup createDefault(BehaviorGroup behaviorGroup) {
 
     @Transactional
     BehaviorGroup create(String accountId, BehaviorGroup behaviorGroup, boolean isDefaultBehaviorGroup) {
+        Long behaviorGroupsSameDisplayNameAlreadyPresent;
+        if (isDefaultBehaviorGroup) {
+            behaviorGroupsSameDisplayNameAlreadyPresent = entityManager.createQuery("SELECT COUNT(*) FROM BehaviorGroup b WHERE b.displayName = :displayName and b.accountId IS NULL", Long.class)
+                    .setParameter("displayName", behaviorGroup.getDisplayName())
+                    .getSingleResult();
+        } else {
+            behaviorGroupsSameDisplayNameAlreadyPresent = entityManager.createQuery("SELECT COUNT(*) FROM BehaviorGroup b WHERE b.displayName = :displayName AND b.accountId = :accountId", Long.class)
+                    .setParameter("displayName", behaviorGroup.getDisplayName())
+                    .setParameter("accountId", accountId)
+                    .getSingleResult();
+        }
+        if (behaviorGroupsSameDisplayNameAlreadyPresent > 0) {
+            throw new IllegalArgumentException(String.format("A behavior group with accountId [%s] and display name [%s] is already present", accountId, behaviorGroup.getDisplayName()));
```

Comment:
What is the HTTP status code returned when this exception is thrown? We need a 400 here and the HTTP response should also give the reason (same message than the exception here). Finally, the 400 status needs to be tested with a RestAssured test.

Reply:
it is a 400. Testing it is a bit tricky as the exception is thrown very deep in the code. I just added a throw new IllegalArgumentException("test"); in the /internal endpoint and this got mapped to a http 400.

Reply:
You could try updating the function at https://github.com/RedHatInsights/notifications-backend/pull/1027/files#diff-59f18caa249dea80d3c35f5a7d8723c28e5c9a5f3c6e820cedbecf807d48e30eR359 to receive the http-code as a param.

Reply:
I think this is what I meant: https://github.com/RedHatInsights/notifications-backend/blob/f23f77df47a905a106c365572f5a4364538c395a/backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java#L369

Passing that value as a param, so you can check the 4xx status in there.

Reply:
ah - actually there are helpers in this class: https://github.com/RedHatInsights/notifications-backend/blob/f23f77df47a905a106c365572f5a4364538c395a/backend/src/test/java/com/redhat/cloud/notifications/CrudTestHelpers.java#L404